service/config_db.hpp

```diff
@@ -0,0 +1,262 @@
+//
+// Created by riz on 29/01/19.
+//
+
+#pragma once
+
+#include <cstring>
+#include <vector>
+#include <sstream>
+#include <sqlite_modern_cpp.h>
+
+namespace raven::db
+{
+  template<typename T>
+  class blob
+  {
+  public:
+    //! getters
+    template<typename vtype>
+    std::enable_if_t<std::is_fundamental<vtype>::value, vtype> get_value() const noexcept
+    {
+        vtype ret;
+        std::memcpy(&ret, data_.data(), sizeof(vtype));
+        return ret;
+    }
+
+    // le compilo me laisse pas faire une template specifier pour std::string je sais pas trop pourquoi ( je suis surement un abrutit mais la il est tard je suis fatigué) à investiguer
+    std::string get_value_string() const noexcept
+    {
+        return (std::string(data_.begin(), data_.end()));
+    }
+
+    const std::vector<T> &get_data() const noexcept
+    {
+        return data_;
+    }
+
+    std::vector<T> &get_data() noexcept
+    {
+        return data_;
+    }
+
+    //! Constructors
+    template<typename vtype, typename std::enable_if_t<std::is_fundamental<T>::value, int> = 0>
+    explicit blob(vtype value) noexcept
+    {
+        data_.resize(sizeof(vtype) / sizeof(T));
+        std::memcpy(data_.data(), &value, sizeof(vtype));
+    }
+
+    explicit blob(std::vector<T> value) noexcept
+    :data_{std::move(value)}
+    {
+    }
+
+    explicit blob(const std::string &value) noexcept
+    :data_{value.begin(), value.end()}
+    {
+    }
+
+    explicit blob(const char *value) noexcept
+    :blob{std::string(value)}
```

Comment:
Seems very under optimised. The memory behind the pointer will be copied once too much.
```suggestion
    :data_{value, value + std::strlen(value)}
```
(may need to add include)

Reply:
True, i just thought of calling the string constructor but this is indeed unoptimized